ares/ws/cartridge/rtc.cpp

```diff
@@ -16,146 +25,245 @@ auto Cartridge::RTC::save() -> void {
 }
 
 auto Cartridge::RTC::tickSecond() -> void {
-  if(++second() < 60) return;
+  static auto bcdIncrement = [](n8& data) -> n8 {
```

Comment:
Do you think perhaps BCD handling could be added to nall?

I ask because as well as this bcdIncrement function, GB TAMA emulation has it's own toBCD/fromBCD functions, 
Nall::CD also contains BCD encode/decode functions used in all CD-ROM based systems.

Perhaps this is a good opportunity to standardise things with a BCD implementation in nall, and updating other places to use it rather than their own.

Reply:
I don't see why not.

Reply:
I'm not sure whether this should be pursued as utility math functions or as a separate type, though; and if it's a type, I'm not sure if I can get the casting to work right (notice how the `second()`, `minute()`, ... functions operate) confidently enough, with my current C++ skillset.

Reply:
We have a `BCD::decode` and `BCD::encode` in n64 as well. I think utility functions are fine, overtyping seems a recipe for more complex code with very little benefit.

Reply:
I'll pick this up after merging, dw about it for now